src/Sylius/Component/User/Security/UserPasswordEncoderInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Lakion package.
+ *
+ * (c) Lakion
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\User\Security;
+
+use Sylius\Component\User\Model\UserInterface;
+
+/**
+ * @author Michał Marcinkowski <michal.marcinkowski@lakion.com>
+ */
+interface UserPasswordEncoderInterface
+{
+    /**
+     * Encodes the user plain password.
```

Comment:
As you said. Im just founding it recursive, that we encode plain password. It will be weird to encode encoded password, DYT? However, If you think that we should leave it here, I'm ok with it.